### DISCUSSION TITLE
Test generator's output

### DISCUSSION THREAD

**Discussion by zohdit** [skhatiri/UAV-Testing-Competition#3] [Test Generation] [answered]
Hi!

When I run the random_generator provided as an example generator, it generates a results folder which contains logs and png files for case studies. Do you expect the same output format for the new test generators? I am not sure if it's fine to save the generated test cases as json files with size and position of obstacles and if this option is already available in Aerialist.

Thanks

**Comment by skhatiri** [comment, marked as answer]
Hi, 

The most important output you need to report is the test case itself, which in Aerialist is represented as a YAML file (similar to the [case studies](https://github.com/skhatiri/UAV-Testing-Competition/blob/master/snippets/case_studies/mission1.yaml)).
- [Here](https://github.com/skhatiri/UAV-Testing-Competition/blob/e3281492673472ad3bbc7a1cefbadb9183c1914a/snippets/testcase.py#L29C1-L30C32) is the code to save the YAML representation of each test case (which also includes the obstacle properties).
- You can also load test case properties from YAML similar to the code [here](https://github.com/skhatiri/UAV-Testing-Competition/blob/e3281492673472ad3bbc7a1cefbadb9183c1914a/snippets/random_generator.py#L14C1-L14C63). You will then have access to the obstacle properties (if any).
